src/test/shared/telemetry/awsTelemetryOptOut.test.ts

```diff
@@ -53,9 +54,10 @@ describe('AwsTelemetryOptOut', () => {
         assert.strictEqual(mockService.telemetryEnabled, false)
     })
 
-    // VS Code has opt-in telemetry by default so this test exists to tell us if Microsoft changes that behavior
-    it('updateTelemetryConfiguration(TelemetryOptOutOptions.SameAsVsCode) enables telemetry', async () => {
+    // VS Code tests use the same preferences as the host editor
+    it('updateTelemetryConfiguration(TelemetryOptOutOptions.SameAsVsCode) matches user setting', async () => {
+        const vscodePreference = !!vscode.workspace.getConfiguration('telemetry').get<boolean>('enableTelemetry')
```

Comment:
This is a unit test rather than an integration test, so we shouldn't hit the 'real' settings. Suggest instead to make `AwsTelemetryOptOut.getVSCodeTelemetrySetting` a constructor parameter, and override it with `() => true` in this test.